packages/rspack-dev-server/src/middleware.ts

```diff
@@ -13,13 +13,26 @@ function etag(buf: any) {
 	return etag;
 }
 
+function createPublicPathGetter(compiler: Compiler) {
+	const raw = compiler.options.output.publicPath || "/";
+
+	if (typeof raw === "function") {
+		return (compilation?: Compilation) =>
+			compilation ? compilation.getPath(raw) : raw({ hash: "XXXX" }, undefined);
+	} else if (/\[(hash|fullhash)[:\]]/.test(raw)) {
+		return (compilation?: Compilation) =>
+			compilation ? compilation.getPath(raw) : raw.replace(/\/$/, "") + "/";
+	} else {
+		return () => raw.replace(/\/$/, "") + "/";
+	}
+}
+
 export function getRspackMemoryAssets(
```

Comment:
This function seems to be obsoleted.

I'm not gonna introduce breaking change in this PR so I didn't remove this function.

Instead, I just made an essential change to it to make tsc succeeded.

Reply:
It's ok to remove this